src/litdata/streaming/item_loader.py

```diff
@@ -136,17 +149,44 @@ def load_item_from_chunk(
             self._chunk_filepaths[chunk_filepath] = True
 
         with open(chunk_filepath, "rb", 0) as fp:
-            fp.seek(offset)
-            pair = fp.read(8)
-            begin, end = np.frombuffer(pair, np.uint32)
-            fp.seek(begin)
-            data = fp.read(end - begin)
+            if encryption and self._config["encryption"]:
+                data = self._load_encrypted_data(fp, offset, encryption)
+            else:
+                data = self._load_data(fp, offset)
 
         # check for mosaic mds format
         if "format" in self._config and self._config["format"] == "mds":
             return self.mds_deserialize(data, chunk_index)
         return self.deserialize(data)
 
+    def _load_encrypted_data(self, fp: Union[FileIO, BytesIO], offset: int, encryption: Encryption) -> bytes:
+        """Load and decrypt data from a file pointer based on the encryption configuration."""
+
+        # Validate the provided encryption object against the expected configuration.
+        self._validate_encryption(encryption)
+
+        # If the encryption level is set to 'chunk', decrypt the entire chunk first.
+        if self._config["encryption"]["level"] == "chunk":
+            encrypted_data = fp.read()
+            decrypted_data = encryption.decrypt(encrypted_data)
```

Comment:
@bhimrazy This loads the entire file every time we read a sample. We should keep the `decrypted_data` in RAM.